### ISSUE TITLE
RRTMG builds failing on dev/12.4.0

### ISSUE THREAD

**Issue by LiamBindle** [geoschem/geos-chem#39]
Hi everyone,

I was testing `feature/CMake` rebased on `dev/12.4.0` and noticed that RRTMG builds were failing for both Make and CMake build systems. 

With the Make build system, I'm trying to compile `geos.mp` from the RRTMG run directories with
```console
$ make mpbuild
```
When I do this I get the following compiler errors
```
ifort -cpp -w -auto -noalign -convert big_endian -O2 -vec-report0 -fp-model source -openmp -mcmodel=medium -shared-intel -traceback -DLINUX_IFORT -DBPCH_DIAG -DBPCH_TIMESER -DBPCH_TPBC -DRRTMG -DNC_HAS_COMPRESSION -DUSE_REAL8 -module ../mod -I/software/include -c rrtmg_rad_transfer_mod.F
rrtmg_rad_transfer_mod.F(788): error #6351: The number of subscripts is incorrect.   [YMID]
       YLAT            = State_Grid%YMid( I, J, 1 )
------------------------------------^
rrtmg_rad_transfer_mod.F(2146): error #6631: A non-optional actual argument must be present when invoking a procedure with an explicit interface.   [STATE_GRID]
      CALL Init_Surface_Rad()
-----------^
rrtmg_rad_transfer_mod.F(2635): error #6404: This name does not have a type, and must have an explicit type.   [STATE_GRID]
      ALLOCATE( CLDFMCL_LW( NGPTLW, State_Grid%NX, State_Grid%NY,
------------------------------------^
rrtmg_rad_transfer_mod.F(2635): error #6460: This is not a field name that is defined in the encompassing structure.   [NX]
      ALLOCATE( CLDFMCL_LW( NGPTLW, State_Grid%NX, State_Grid%NY,
-----------------------------------------------^
rrtmg_rad_transfer_mod.F(2635): error #6385: The highest data type rank permitted is INTEGER(KIND=8).   [NX]
      ALLOCATE( CLDFMCL_LW( NGPTLW, State_Grid%NX, State_Grid%NY,
-----------------------------------------------^
rrtmg_rad_transfer_mod.F(2635): error #6460: This is not a field name that is defined in the encompassing structure.   [NY]
      ALLOCATE( CLDFMCL_LW( NGPTLW, State_Grid%NX, State_Grid%NY,
--------------------------------------------------------------^
rrtmg_rad_transfer_mod.F(2635): error #6385: The highest data type rank permitted is INTEGER(KIND=8).   [NY]
      ALLOCATE( CLDFMCL_LW( NGPTLW, State_Grid%NX, State_Grid%NY,
--------------------------------------------------------------^
rrtmg_rad_transfer_mod.F(2636): error #6460: This is not a field name that is defined in the encompassing structure.   [NZ]
     &                      State_Grid%NZ ), STAT=AS )
---------------------------------------^
rrtmg_rad_transfer_mod.F(2636): error #6385: The highest data type rank permitted is INTEGER(KIND=8).   [NZ]
     &                      State_Grid%NZ ), STAT=AS )
---------------------------------------^

.
.
.

/tmp/ifortbySTCx.i(2802): catastrophic error: Too many errors, exiting
compilation aborted for rrtmg_rad_transfer_mod.F (code 1)
```
And I get this for all 4 RRTMG run directories.

Are there any changes to how I should be compiling GC+RRTMG in `12.4.0`?

Thanks,

Liam

**Comment by yantosca** [comment]
Thanks for reporting this.  So State_Grid%YMID is now a 2-D array.  The old GET_YMID did take a I,J,L arguments before, so this is probably just a cut-n-paste issue.   Because we don't usually test the RRTMG simulation, we didn't discover this.  In practice only a couple of users in the GC community are running with it.

Also -- FYI, I have not gotten RRTMG to compile with gfortran, it chokes with an internal compiler error.  Because RRTMG is 3rd-party code I didn't even try to modify it.   The RRTMG code that we have is probably older than the latest version available from AER, but no word if they have gotten it to work with gfortran.

A while back I also started adding RRTMG netCDF diagnostics but I did not really get a chance to validate them -- we would need help from someone in the Aerosols WG for that.
